From 3f9f0e076de72c6121ba95fb000072949be6cf7e Mon Sep 17 00:00:00 2001
From: Alexandru Gagniuc <mr.nuke.me@gmail.com>
Date: Sat, 10 May 2025 20:37:03 +0200
Subject: [PATCH] nss-dp: Fix build falures due to -Wmissing-prototypes

Some functions are only used within the same file, and sould thus be
declared static. For others, the correct header wasn't included. These
error are exposed by the "-Wmissing-prototypes" compiler flag.

Mark static functions as static, and include the correct headers for
global functions.

Signed-off-by: Alexandru Gagniuc <mr.nuke.me@gmail.com>
---
 hal/dp_ops/edma_dp/edma_v1/edma_cfg.c |  2 +-
 hal/gmac_ops/qcom/qcom_if.c           |  4 ++--
 nss_dp_main.c                         | 10 +++++++---
 3 files changed, 10 insertions(+), 6 deletions(-)

--- a/hal/dp_ops/edma_dp/edma_v1/edma_cfg.c
+++ b/hal/dp_ops/edma_dp/edma_v1/edma_cfg.c
@@ -714,7 +714,7 @@ static void edma_configure_rings(struct
  * edma_hw_reset()
  *	Reset EDMA Hardware during initialization
  */
-int edma_hw_reset(struct edma_hw *ehw)
+static int edma_hw_reset(struct edma_hw *ehw)
 {
 	struct reset_control *rst;
 	struct platform_device *pdev = ehw->pdev;
--- a/hal/gmac_ops/qcom/qcom_if.c
+++ b/hal/gmac_ops/qcom/qcom_if.c
@@ -242,7 +242,7 @@ static int32_t qcom_get_netdev_stats(str
  * qcom_get_strset_count()
  *	Get string set count for ethtool operations
  */
-int32_t qcom_get_strset_count(struct nss_gmac_hal_dev *nghd, int32_t sset)
+static int32_t qcom_get_strset_count(struct nss_gmac_hal_dev *nghd, int32_t sset)
 {
 	struct net_device *netdev = nghd->netdev;
 
@@ -261,7 +261,7 @@ int32_t qcom_get_strset_count(struct nss
  * qcom_get_strings()
  *	Get strings
  */
-int32_t qcom_get_strings(struct nss_gmac_hal_dev *nghd, int32_t sset,
+static int32_t qcom_get_strings(struct nss_gmac_hal_dev *nghd, int32_t sset,
 						uint8_t *data)
 {
 	struct net_device *netdev = nghd->netdev;
--- a/nss_dp_main.c
+++ b/nss_dp_main.c
@@ -16,6 +16,8 @@
  * OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
+#include "nss_dp_dev.h"
+
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/if_vlan.h>
@@ -782,7 +784,7 @@ bool nss_dp_is_phy_dev(struct net_device
 /*
  * nss_dp_adjust_link()
  */
-void nss_dp_adjust_link(struct net_device *netdev)
+static void nss_dp_adjust_link(struct net_device *netdev)
 {
 	struct nss_dp_dev *dp_priv = netdev_priv(netdev);
 	int current_state = dp_priv->link_state;
@@ -1085,6 +1087,7 @@ int32_t nss_dp_get_port_num(struct net_d
 }
 EXPORT_SYMBOL(nss_dp_get_port_num);
 
+#ifdef NSS_DP_PPEDS_SUPPORT
 /*
  * nss_dp_ppeds_get_ops()
  *	API to get PPE-DS operations
@@ -1094,6 +1097,7 @@ struct nss_dp_ppeds_ops *nss_dp_ppeds_ge
 	return nss_dp_ppeds_ops_get();
 }
 EXPORT_SYMBOL(nss_dp_ppeds_get_ops);
+#endif
 
 /*
  * nss_dp_nsm_sawf_sc_stats_read()
@@ -1108,7 +1112,7 @@ EXPORT_SYMBOL(nss_dp_nsm_sawf_sc_stats_r
 /*
  * nss_dp_init()
  */
-int __init nss_dp_init(void)
+static int __init nss_dp_init(void)
 {
 	int ret, i;
 
@@ -1191,7 +1195,7 @@ int __init nss_dp_init(void)
 /*
  * nss_dp_exit()
  */
-void __exit nss_dp_exit(void)
+static void __exit nss_dp_exit(void)
 {
 	platform_driver_unregister(&nss_dp_drv);
 
